sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/PartitionControllerImpl.java

```diff
@@ -109,56 +109,60 @@ public PartitionControllerImpl(
     }
 
     private Mono<Void> removeLease(Lease lease) {
-            if (this.currentlyOwnedPartitions.get(lease.getLeaseToken()) != null) {
-                WorkerTask workerTask = this.currentlyOwnedPartitions.remove(lease.getLeaseToken());
-
+        return Mono.justOrEmpty(this.currentlyOwnedPartitions.remove(lease.getLeaseToken()))
```

Comment:
use Mono.just(this) to handle the WorkTask removal from the concurrent list and pipe it to a flatMap that will do the lease update (see leaseManager.release()).

Reply:
Chatted offline, updated.